source/extensions/path/uri_template_lib/uri_template_internal.cc

```diff
@@ -46,6 +47,14 @@ constexpr absl::string_view kLiteral = "a-zA-Z0-9-._~" // Unreserved
                                        ":@"
                                        "="; // user included "=" allowed
 
+// Additional literal that allows "*" in the pattern.
+// Valid pchar from https://datatracker.ietf.org/doc/html/rfc3986#appendix-A
+constexpr absl::string_view kLiteralAsterisk = "a-zA-Z0-9-._~" // Unreserved
```

Comment:
nit: This should probably be "LiteralWithAsterisk". The current name makes it sound like it would be just a literal asterisk. (And perhaps add a comment about removing when we remove the runtime feature)